File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        int answer[] = new int[n];
        k=k%n;
        for(int i =0;i<n;i++){
          int newIndex= (k+i)%n;
          answer[newIndex] = nums[i];
          newIndex++;
        }

        for(int i = 0;i<n;i++){
            nums[i]= answer[i];
        }
    }
}

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
       n=len(nums)
       k=k%n
       result = [0]*n
       for i in range(len(nums)):
        newindex=(k+i)%n
        result[newindex]=nums[i]
      
       for i in range(len(nums)):
        nums[i] = result[i]

        